samples/echoBot/main.go

```diff
@@ -45,13 +50,20 @@ func main() {
 		MaxRoutines: ext.DefaultMaxRoutines,
 	})
 	updater := ext.NewUpdater(dispatcher, nil)
+	go func() {
+		// sentinel, when the context is done, stop all bots.
+		<-ctx.Done()
+		updater.Stop()
+		updater.StopAllBots()
```

Comment:
Initially just called StopAllBots, but that wasn't enough to stop updater from Idle()'ing.

Reply:
Hmm. that might be worth discussing in itself!